packages/server/src/api/routes/tests/search.spec.ts

```diff
@@ -67,6 +67,22 @@ describe.each([
   class SearchAssertion {
     constructor(private readonly query: RowSearchParams) {}
 
+    private findRow(expectedRow: any, foundRows: any[]) {
+      const row = foundRows.find(foundRow => _.isMatch(foundRow, expectedRow))
+      if (!row) {
+        const fields = Object.keys(expectedRow)
+        // To make the error message more readable, we only include the fields
+        // that are present in the expected row.
+        const searchedObjects = foundRows.map(row => _.pick(row, fields))
+        throw new Error(
+          `Failed to find row: ${JSON.stringify(
+            expectedRow
+          )} in ${JSON.stringify(searchedObjects)}`
+        )
+      }
+      return row
+    }
```

Comment:
Added this so that when tests fail because the row searched for does not match any of the rows returned, the message is _much_ nicer. It used to complain about passing `undefined` to `objectContaining`, now it looks like this:

![CleanShot 2024-05-03 at 17 35 21@2x](https://github.com/Budibase/budibase/assets/338027/6634dbe5-d3b2-4ffe-abdd-7bb1ad0b3c14)